mutation_compactor.hh

```diff
@@ -179,8 +179,8 @@ private:
     stop_iteration do_consume(range_tombstone_change&& rtc, Consumer& consumer, GCConsumer& gc_consumer) {
         stop_iteration gc_consumer_stop = stop_iteration::no;
```

Comment:
patch changelog should explain the solution.

Reply:
Updated the commit message with the explanation matching the PR description.